docs/src/rules/no-invalid-this.md

```diff
@@ -284,6 +284,60 @@ obj.Foo = function Foo() {
 
 :::
 
+This rule additionally supports TypeScript type syntax.
+
+Examples of **incorrect** TypeScript code for this rule:
+
+:::incorrect
+
+```ts
+/*eslint no-invalid-this: "error"*/
+
+interface SomeType {
+    prop: string;
+}
+
+function foo(bar: string) {
+    this.prop;
+    console.log(bar)
+}
+
+/** @this Obj */
+foo(function() {
+    console.log(this);
+    z(x => console.log(x, this));
+});
+
+```
+
+:::
+
+Examples of **correct** TypeScript code for this rule:
+
+:::correct
+
+```ts
+/*eslint no-invalid-this: "error"*/
+
+interface SomeType {
+    prop: string;
+}
+
+function foo(this: SomeType) {
+    this.prop;
+}
+
+class A {
+    a = 5;
+    b = this.a;
+    accessor c = this.a;
+}
+```
+
+:::
+
 ## When Not To Use It
 
 If you don't want to be notified about usage of `this` keyword outside of classes or class-like objects, you can safely disable this rule.
+
+The TypeScript compiler already checks for the issues addressed by this ESLint rule. Therefore, enabling this rule in new TypeScript projects is generally unnecessary. You should only enable it if you prefer ESLint's error messages over the TypeScript compiler's or if you're using it to lint JavaScript code.
```

Comment:
We already have meta fields for this purpose. For example:

https://github.com/eslint/eslint/blob/6be36c99432ecdc72e33b6fb3293cf28f66ab78d/docs/src/rules/no-import-assign.md?plain=1#L4-L5

Reply:
Actually, this rule's docs already have these fields specified, so this paragraph seems like an unnecessary duplicate.

https://eslint.org/docs/latest/rules/no-invalid-this#handled_by_typescript